includes/RemovePIIJob.php

```diff
@@ -349,6 +346,8 @@ public function run() {
 			if ( $dbw->tableExists( $key, __METHOD__ ) ) {
 				foreach ( $value as $name => $fields ) {
 					try {
+						$lbFactory->waitForReplication();
```

Comment:
you'll want to wait for replication lag *after* performing actions that cause the lag (writes) instead of before it

Reply:
Done